cmd/insprd/auth/jwt/auth.go

```diff
@@ -0,0 +1,100 @@
+// Package jwtauth is responsible for implementing the auth
+// methods specified in the auth folder of the inspr pkg.
+package jwtauth
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/lestrrat-go/jwx/jwt"
+	"gitlab.inspr.dev/inspr/core/pkg/auth"
+	"gitlab.inspr.dev/inspr/core/pkg/auth/models"
+	"gitlab.inspr.dev/inspr/core/pkg/ierrors"
+	"gitlab.inspr.dev/inspr/core/pkg/rest/request"
+)
+
+// JWTauth implements the Auth interface for jwt authetication provider
+type JWTauth struct{}
+
+// NewJWTauth returns a JWTAuth object
+func NewJWTauth() *JWTauth {
+	return &JWTauth{}
+}
+
+//Validate receives the
+func (JA *JWTauth) Validate(token []byte) (models.Payload, []byte, error) {
+	jwtToken, err := jwt.Parse(token, jwt.WithValidate(true))
+	// not valid token
+	if err != nil {
+		return models.Payload{}, token, err
+	}
+
+	expiration, found := jwtToken.Get(jwt.ExpirationKey)
+	now := time.Now()
+
+	// expired
+	if now.After(expiration.(time.Time)) || !found {
+		newToken, err := JA.Refresh(token)
+		if err != nil {
+			return models.Payload{},
+				token,
+				errors.New("error refreshing token")
+		}
+		token = newToken
+	}
+
+	// gets payload from token
+	payload, found := jwtToken.Get("payload")
+	if !found {
+		return models.Payload{},
+			token,
+			errors.New("payload not found")
+	}
+
+	return payload.(models.Payload), token, nil
+}
+
+// Tokenize receives a payload and returns it in signed jwt format. Uses JWT authentication provider
+func (JA *JWTauth) Tokenize(load models.Payload) ([]byte, error) {
+
+	URL := os.Getenv("AUTH_PATH")
```

Comment:
Shouldn't there be an error if this env var is empty/doesn't exists?  

If so, it might be a good idea to add this in the struct JWTauth so it's existence is validated when instantiating a new structure